Problem: `make rpm` command requires sudo access

[Install](https://github.com/Seagate/hare/blob/master/Makefile#L124) target in Makefile calls [$(DESTDIR)/var/lib/hare](https://github.com/Seagate/hare/blob/master/Makefile#L141). This target runs comannd `groupadd` command which requires sudoer permission to run. So `make rpm` requires sudo permission.

Solution:
`install` target should not contain commands that require sudoers permission. As a user without sudoer permission should also be able to run `install` target.  
 
 ```
creating build/bdist.linux-x86_64/wheel/hax-0.1.0.dist-info/WHEEL
creating 'dist/hax-0.1.0-cp36-cp36m-linux_x86_64.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'libhax.cpython-36m-x86_64-linux-gnu.so'
adding 'hax/__init__.py'
adding 'hax/exception.py'
adding 'hax/ffi.py'
adding 'hax/halink.py'
adding 'hax/handler.py'
adding 'hax/hax.py'
adding 'hax/message.py'
adding 'hax/server.py'
adding 'hax/types.py'
adding 'hax/util.py'
adding 'hax-0.1.0.dist-info/METADATA'
adding 'hax-0.1.0.dist-info/WHEEL'
adding 'hax-0.1.0.dist-info/entry_points.txt'
adding 'hax-0.1.0.dist-info/top_level.txt'
adding 'hax-0.1.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
--> Checking cfgen
--> Checking hax
--> Checking files with mypy
--> Checking files with flake8
running flake8
running mypy
Success: no issues found in 1 source file
Success: no issues found in 1 source file
Success: no issues found in 1 source file
Success: no issues found in 10 source files

+ exit 0
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.6jsVUR
+ umask 022
+ cd /home/vagrant/rpmbuild/BUILD
+ '[' /home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64 '!=' / ']'
+ rm -rf /home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64
++ dirname /home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64
+ mkdir -p /home/vagrant/rpmbuild/BUILDROOT
+ mkdir /home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64
+ cd hare
+ rm -rf /home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64
+ make install DESTDIR=/home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64
install: creating directory '/home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64'
install: creating directory '/home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64/var'
install: creating directory '/home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64/var/lib'
install: creating directory '/home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64/var/lib/hare'
bash: /usr/sbin/groupadd: Permission denied
make[2]: *** [/home/vagrant/rpmbuild/BUILDROOT/hare-0.1.0-1_git8995cba_m0gitfffc3996c.el7.x86_64/var/lib/hare] Error 126
error: Bad exit status from /var/tmp/rpm-tmp.6jsVUR (%install)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.6jsVUR (%install)
make[1]: *** [__rpm] Error 1
make: *** [rpm] Error 2

```

--- a/Makefile
+++ b/Makefile
@@ -138,14 +138,8 @@ install: install-dirs install-cfgen install-hax install-systemd install-vendor
 	@install --verbose --directory $(DESTDIR)/usr/bin
 	@ln -sf /$(PREFIX)/bin/hctl $(DESTDIR)/usr/bin
 
-$(DESTDIR)/var/lib/hare:
-	@install --verbose --directory --mode=0775 $@
-	@groupadd --force hare
-	@chgrp hare $@
-	@chmod --changes g+w $@
-
 .PHONY: install-dirs
-install-dirs: $(DESTDIR)/var/lib/hare
+install-dirs:
 	@for d in $(HARE_CONF) \
                   $(HARE_LIBEXEC) \
 	          $(DESTDIR)/var/log/hare \
@@ -153,6 +147,7 @@ install-dirs: $(DESTDIR)/var/lib/hare
 	 do \
 	     install --verbose --directory $$d; \
 	 done
+	@install --verbose --directory --mode=0775 $(DESTDIR)/var/lib/hare
 
 .PHONY: unpack-dhall-prelude
 unpack-dhall-prelude:
@@ -222,6 +217,12 @@ devinstall: install-dirs devinstall-cfgen devinstall-hax devinstall-systemd devi
 	@$(call _log,linking hctl -> $(DESTDIR)/usr/bin)
 	@install --verbose --directory $(DESTDIR)/usr/bin
 	@ln -sf $(TOP_SRC_DIR)hctl $(DESTDIR)/usr/bin
+	@$(call _log,creating hare group)
+	@groupadd --force hare
+	@$(call _log,changing permission of $(DESTDIR)/var/lib/hare)
+	@chgrp hare $(DESTDIR)/var/lib/hare 
+	@chmod --changes g+w $(DESTDIR)/var/lib/hare
+
 
 .PHONY: devinstall-cfgen
 devinstall-cfgen: CFGEN_INSTALL_CMD = ln -sf

--- a/hare.spec
+++ b/hare.spec
@@ -63,7 +63,7 @@ systemctl daemon-reload
 install --directory --mode=0775 /var/lib/hare
 groupadd --force hare
 chgrp hare /var/lib/hare
-chmod g+w /var/lib/hare
+chmod --changes g+w /var/lib/hare
 
 %postun
 systemctl daemon-reload

